presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -221,6 +224,29 @@ protected void setup(Binder binder)
             config.setKeyStorePassword(internalCommunicationConfig.getKeyStorePassword());
         });
 
+        if (internalCommunicationConfig.isKerberosEnabled()) {
+            File kerberosConfig = internalCommunicationConfig.getKerberosConfig();
+            File kerberosKeytab = internalCommunicationConfig.getKerberosKeytab();
+            String kerberosPrincipal = internalCommunicationConfig.getKerberosPrincipal();
+            String kerberosServiceName = internalCommunicationConfig.getKerberosServiceName();
+            checkArgument(kerberosConfig != null, "kerberos config must be set");
+            checkArgument(kerberosKeytab != null, "kerberos keytab must be set");
+            checkArgument(kerberosPrincipal != null, "kerberos principal must be set");
+            checkArgument(kerberosServiceName != null, "kerberos service name must be set");
+
+            configBinder(binder).bindConfigGlobalDefaults(KerberosConfig.class, config -> {
```

Comment:
how does it relate to `KerberosConfig` configured in com/facebook/presto/server/security/ServerSecurityModule.java:50

What will happen if internal kerberos communication does not match server kerboros communication? WIll one overwrite the other?